docs/troubleshooting-guide.md

```diff
@@ -0,0 +1,262 @@
+
+## Troubleshooting: Installing & Running Packages
+
+### Issue 1: can't install a dependency (which is a local package)
+
+Usually occurs when running `lerna bootstrap` immediately after a clean install.
+
+Example error message:
+```
+lerna ERR! npm install exited 1 in '@zooniverse/react-components'
+lerna ERR! npm install stderr:
+npm ERR! code E404
+npm ERR! 404 Not Found: @zooniverse/grommet-theme@2.0.0
+```
+
+Analysis:
+- This happens because Package A (react-components) requires Package B
+  (grommet-theme), and in this case, Package B needs to be **built** first
+  before Package A can use it.
+
+Solution:
+- Build Package B first, e.g.:
+  - `cd packages/grommet-theme`
+  - `lerna bootstrap --scope=@zooniverse/grommet-theme`
+  - `npm run build`
+- Run `lerna link` to 'register' the newly built package.
+
+### Issue 2: when running script, can't find installed dependency
+
+Occurs when running `npm run dev` or similar script from within a package.
+Despite the fact that the dependencies are clearly installed in the package's
+`/node_modules/`, they're not being registered.
+
+Example error message:
+```
+$ cd packages/lib-classifier
+$ npm run _check
+
+> @zooniverse/classifier@0.0.1 _check /Users/shaun/projects/front-end-monorepo/packages/lib-classifier
+> check-dependencies
+
+sh: check-dependencies: command not found
+npm ERR! file sh
+npm ERR! code ELIFECYCLE
+npm ERR! errno ENOENT
+npm ERR! syscall spawn
+npm ERR! @zooniverse/classifier@0.0.1 _check: `check-dependencies`
+npm ERR! spawn ENOENT
+npm ERR! 
+npm ERR! Failed at the @zooniverse/classifier@0.0.1 _check script.
+```
+
+Analysis
+- Did you try installing dependecies for the package `npm install`?
+- **Do NOT use** `npm install`. This tends to break all sorts of links between
+  the packages.
+
+Solution:
+- Instead of using `npm install`, install the dependencies for the package using
+  `lerna bootstrap --scope=my-package`
+  - If necessary, remove the package's existing `node_modules` first.
+- If necessary, re-run `lerna bootstrap` to re-establish the links
+  between all packages.
+
+### Issue 3: system lag or memory leak during/after `lerna bootstrap`
+
+During, or shortly after, running `lerna bootstrap`, you might encounter bizarre
+situations where the computer starts lagging, or running a script results in
+unexpected errors that talks about "overflows" or "too many files open at once"
+or "out of memory".
+
+Example error message:
+```
+$ cd packages/lib-classifier/
+$ npm run dev
+=> FAILURE: Error: ENFILE: file table overflow, open '/Users/shaun/.nvm/versions/node/v8.2.1/lib/node_modules/npm/node_modules/update-notifier/node_modules/import-lazy/index.js'
+```
+
+Analysis:
+- Yeah... this sort of thing happens randomly, but not so rarely that it's not
+  worth noting.
+- It's nothing to do with the Monorepo code, it's just a quirk of the computer.
+
+Solution:
+- Try again after 30 seconds
+- Restart your computer
+
+I wish I was kidding.
```

Comment:
Oh no! I wonder if this has to do with hardware specs or OS version? I've seen lerna be a bit slow, but I haven't seen this particular error 🤔